packages/react-draggable-dialog/src/components/DraggableDialog/DraggableDialog.tsx

```diff
@@ -1,8 +1,91 @@
 import * as React from 'react';
-import { mergeClasses } from '@fluentui/react-components';
-import { useStyles } from './DraggableDialog.styles';
+import {
+  DndContext,
+  useSensor,
+  MouseSensor,
+  TouchSensor,
+  KeyboardSensor,
+  useSensors,
+  DragEndEvent,
+  PointerSensor,
+} from '@dnd-kit/core';
+import { Dialog, useId } from '@fluentui/react-components';
 
-export const DraggableDialog: React.FC = () => {
-  const styles = useStyles();
-  return <div className={mergeClasses(styles.root)}>Hello World!</div>;
+import { DraggableDialogContextProvider } from '../../contexts/DraggableDialogContext';
+import { DraggableDialogProps } from './DraggableDialog.types';
+import { getParsedDraggableMargin, restrictToMarginModifier } from './utils';
+
+export const DraggableDialog: React.FC<DraggableDialogProps> = (props) => {
+  const { margin, keepInViewport, id, announcements } = {
+    id: useId('draggable-dialog-'),
+    keepInViewport: true,
+    ...props,
+    margin: getParsedDraggableMargin(props.margin),
+  };
+
+  const [isDragging, setIsDragging] = React.useState(false);
+  const [hasBeenDragged, setHasBeenDragged] = React.useState(false);
+  const [position, setPosition] = React.useState({ x: 0, y: 0 });
+
+  const mouseSensor = useSensor(MouseSensor);
+  const pointerSensor = useSensor(PointerSensor);
+  const touchSensor = useSensor(TouchSensor);
+  const keyboardSensor = useSensor(KeyboardSensor);
+  const sensors = useSensors(
+    pointerSensor,
+    mouseSensor,
+    touchSensor,
+    keyboardSensor
+  );
+
+  const onDragEnd = React.useCallback((event: DragEndEvent) => {
+    setPosition(({ x, y }) => ({
+      x: x + event.delta.x,
+      y: y + event.delta.y,
+    }));
+    setIsDragging(false);
+  }, []);
+
+  const onDragStart = React.useCallback(() => {
+    setHasBeenDragged(true);
+    setIsDragging(true);
+  }, []);
+
+  const restrictToMargin = React.useMemo(() => {
+    return restrictToMarginModifier({ margin, keepInViewport });
+  }, [margin, keepInViewport]);
+
+  const dndAnnouncements = React.useMemo(() => {
+    if (!announcements) {
+      return;
+    }
+
+    return {
+      onDragStart: () => announcements.start,
+      onDragEnd: () => announcements.end,
+    };
+  }, [announcements]);
+
+  return (
+    <DndContext
+      sensors={sensors}
+      modifiers={[restrictToMargin]}
```

Comment:
haven't check the latest development in performance for jsx rendering, but AFAIR these inline objects will be re-created on every render, and because we are instantiating context all children using the context included. can we improve this ?

Reply:
Fixed!